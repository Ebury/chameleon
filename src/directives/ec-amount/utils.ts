
import defaultOptions from './options';
import type { AmountDirectiveOptions } from './types';

function format(input: undefined, opts?: AmountDirectiveOptions): undefined;
function format(input: null, opts?: AmountDirectiveOptions): null;
function format(input: string | number, opts?: AmountDirectiveOptions): string;
function format(input: string | number | undefined | null, opts: AmountDirectiveOptions = defaultOptions): string | null | undefined {
  /* eslint-disable no-param-reassign */
  if (input === '' || input === null || input === undefined) {
    return input;
  }

  if (typeof input === 'number') {
    input = input.toString();
  }

  const negative = input.startsWith('-') ? '-' : '';

  input = sanitizeInput(input, opts.decimalSeparator);
  if (opts.precision > 0) {
    input = keepOnlyOneDecimalSeparator(input, opts.decimalSeparator);
  } else {
    input = removeDecimal(input, opts.decimalSeparator);
  }
  const parts = input.split(opts.decimalSeparator);
  let integralPart = parseToStr(parts[0]);
  let fractionalPart = parts[1];

  if (fractionalPart && fractionalPart.length > opts.precision) {
    fractionalPart = fractionalPart.substring(0, opts.precision);
  }

  integralPart = addGroupingSeparators(integralPart, opts.groupingSeparator);
  const result = negative + joinIntegralAndFractionalParts(integralPart, fractionalPart, opts.decimalSeparator);
  return result;
}

function parseToStr(number: string): string {
  const parsed = Number.parseFloat(number);
  if (Number.isNaN(parsed)) {
    return '';
  }
  return parsed.toString();
}

function keepOnlyOneDecimalSeparator(number: string, decimalSeparator: string): string {
  if (number.includes(decimalSeparator)) {
    const split = number.split(decimalSeparator);
    return split[0] + decimalSeparator + split.slice(1).join('');
  }
  return number;
}

function removeDecimal(number: string, decimalSeparator: string): string {
  const split = number.split(decimalSeparator);
  return split[0];
}

function unFormat(input: string, groupingSeparator: string, decimalSeparator: string): string {
  if (!input) {
    return input;
  }

  return input
    .replace(new RegExp(`\\${groupingSeparator}`, 'g'), '')
    .replace(new RegExp(`\\${decimalSeparator}`, 'g'), '.');
}

function sanitizeInput(input: string, decimalSeparator: string): string {
  return input.replace(new RegExp(`[^0-9${decimalSeparator}]`, 'g'), '') || '';
}

function addGroupingSeparators(integralPart: string, separator: string): string {
  return integralPart.replace(/(\d)(?=(?:\d{3})+\b)/gm, `$1${separator}`);
}

function joinIntegralAndFractionalParts(integralPart: string, fractionalPart: string, decimalSeparator: string): string {
  return fractionalPart !== undefined ? integralPart + decimalSeparator + fractionalPart : integralPart;
}

/* c8 ignore start */
function setCursor(el: HTMLInputElement, position: number) {
  const setSelectionRange = function setSelectionRange() {
    el.setSelectionRange(position, position);
  };

  if (el === document.activeElement) {
    setSelectionRange();
    setTimeout(setSelectionRange, 1); // Android Fix
  }
}
/* c8 ignore stop */

export {
  format,
  sanitizeInput,
  setCursor,
  unFormat,
};
