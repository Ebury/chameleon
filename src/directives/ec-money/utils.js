import defaults from './options';

function format(input, opt = defaults) {
  if (input === '' || input === null || input === undefined) {
    return input;
  }

  if (typeof input === 'number') {
    input = input.toString();
  }

  const negative = input.includes('-') ? '-' : '';

  input = sanitizeInput(input, opt.decimalSeparator);
  input = keepOnlyOneDecimalSeparator(input, opt.decimalSeparator);
  const parts = input.split(opt.decimalSeparator);
  let integer = parseToStr(parts[0]);
  let decimal = parts[1];

  if (decimal && decimal.length > opt.precision) {
    decimal = decimal.substr(0, opt.precision);
  }

  integer = addGroupingSeparators(integer, opt.groupingSeparator);
  const result = negative + joinIntegerAndDecimal(integer, decimal, opt.decimalSeparator);
  return result;
}

function parseToStr(number) {
  /* eslint no-restricted-globals: "off" */
  const parsed = parseFloat(number);
  if (isNaN(parsed)) {
    return '';
  }
  return parsed.toString();
}

function keepOnlyOneDecimalSeparator(number, decimalSeparator) {
  if (number.includes(decimalSeparator)) {
    const split = number.split(decimalSeparator);
    return split[0] + decimalSeparator + split.slice(1).join('');
  }
  return number;
}

function unFormat(input, groupingSeparator, decimalSeparator) {
  /* eslint no-param-reassign: "off" */
  input = (`${input}`).replace(new RegExp(`\\${groupingSeparator}`, 'g'), '');
  input = (`${input}`).replace(new RegExp(`\\${decimalSeparator}`, 'g'), '.');
  return input;
}

function sanitizeInput(input, decimalSeparator) {
  return input.replace(new RegExp(`[^0-9${decimalSeparator}]`, 'g'), '') || '';
}

function addGroupingSeparators(integer, separator) {
  return integer.replace(/(\d)(?=(?:\d{3})+\b)/gm, `$1${separator}`);
}

function joinIntegerAndDecimal(integer, decimal, decimalSeparator) {
  return decimal !== undefined ? integer + decimalSeparator + decimal : integer;
}

function setCursor(el, position) {
  const setSelectionRange = function setSelectionRange() {
    el.setSelectionRange(position, position);
  };

  if (el === document.activeElement) {
    setSelectionRange();
    setTimeout(setSelectionRange, 1); // Android Fix
  }
}

// https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Creating_and_triggering_events#The_old-fashioned_way
function event(name) {
  const evt = document.createEvent('Event');
  evt.initEvent(name, true, true);
  return evt;
}

export {
  format,
  unFormat,
  setCursor,
  event,
  sanitizeInput,
};
