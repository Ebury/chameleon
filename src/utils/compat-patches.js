/* istanbul ignore file */

import {
  Dropdown as FvDropdown,
  TooltipDirective as FvTooltipDirective,
  VClosePopper,
  VTooltip,
} from 'floating-vue';

// we need to tell the @vue/compat that all components coming from floating-vue are Vue 3 components so it skips
// them when patching component instances. These components can't run in default mode for Vue 2, there is lots of
// errors coming from the console in that mode.
export default function applyCompatPatches() {
  const compatConfig = {
    MODE: 3,
    RENDER_FUNCTION: false,
  };

  VTooltip.compatConfig = compatConfig;
  VClosePopper.compatConfig = compatConfig;
  FvTooltipDirective.compatConfig = compatConfig;
  for (const component of Object.values(FvTooltipDirective.components)) {
    component.compatConfig = compatConfig;
  }

  FvDropdown.compatConfig = compatConfig;

  for (const component of Object.values(FvDropdown.components)) {
    component.compatConfig = compatConfig;
  }
}
