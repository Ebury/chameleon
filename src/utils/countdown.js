import mitt from 'mitt';

export default class Countdown {
  constructor() {
    const m = mitt();
    Object.assign(this, m);
  }

  reduceMinutesLeft(totalSeconds) {
    this.getminutes(totalSeconds);
    this.getseconds(totalSeconds, this.minutesLeft);
  }

  start(secondsToGo, showMinutes) {
    this.seconds = secondsToGo;
    this.secondsLeft = secondsToGo;
    this.currentTime = Math.ceil(Date.now() / 1000);
    this.startTime = Math.ceil(Date.now() / 1000);
    this.interval = setInterval(() => this.reduceSecondsLeft(showMinutes), 1000);
  }

  get timeDifference() {
    return this.currentTime - this.startTime;
  }

  reduceSecondsLeft(showMinutes) {
    this.currentTime = Math.ceil(Date.now() / 1000);
    this.totalSeconds = this.seconds - this.timeDifference;
    this.minutesLeft = null;
    this.secondsLeft = this.totalSeconds;

    if (showMinutes) {
      this.reduceMinutesLeft(this.totalSeconds);
    }

    this.emit('time-updated', {
      seconds: this.secondsLeft,
      minutes: this.minutesLeft,
      totalSeconds: this.totalSeconds,
    });

    if (this.totalSeconds <= 0) {
      this.stop();
      this.emit('time-expired');
    }
  }

  stop() {
    clearInterval(this.interval);
  }

  getminutes(totalSeconds) {
    this.minutesLeft = Math.floor(totalSeconds / 60);
  }

  getseconds(totalSeconds, minutes) {
    // take mins remaining (as seconds) away from total seconds remaining
    this.secondsLeft = totalSeconds - Math.round(minutes * 60);
  }
}
