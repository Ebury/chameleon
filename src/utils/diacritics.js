export function removeDiacritics(str) {
  if (typeof str === 'string' && str.normalize) {
    // https://stackoverflow.com/questions/990904/remove-accents-diacritics-in-a-string-in-javascript
    return str.normalize('NFD').replace(/[\u0300-\u036f]/g, '');
  }

  // IE11 users will not enjoy this feature:
  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize
  return str;
}
