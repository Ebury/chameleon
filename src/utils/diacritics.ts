export function removeDiacritics(str: string): string {
  if (str && str.normalize) {
    // https://stackoverflow.com/questions/990904/remove-accents-diacritics-in-a-string-in-javascript
    return str.normalize('NFD').replace(/[\u0300-\u036f]/g, '');
  }

  // IE11 users will not enjoy this feature:
  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize
  return str;
}
