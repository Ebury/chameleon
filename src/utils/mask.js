export function mask(text, maskSymbol = '*', visibleChars = 2) {
  if (!text) {
    throw new Error('Text is required');
  }
  if (!maskSymbol) {
    throw new Error('Mask symbol cannot be empty');
  }
  if (!visibleChars || visibleChars < 0) {
    throw new Error('Visible characters must be a number greater than zero');
  }

  const digitsToBeMasked = text.length - visibleChars;
  return maskSymbol.repeat(digitsToBeMasked) + text.substr(digitsToBeMasked);
}
