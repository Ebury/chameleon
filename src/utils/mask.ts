export function mask(text: string, maskSymbol: string = '*', visibleChars: number = 2): string {
  if (!text) {
    throw new Error('Text is required');
  }
  if (!maskSymbol) {
    throw new Error('Mask symbol cannot be empty');
  }
  if (!visibleChars || visibleChars < 0) {
    throw new Error('Visible characters must be a number greater than zero');
  }

  const digitsToBeMasked = text.length - visibleChars;
  return maskSymbol.repeat(digitsToBeMasked) + text.substring(digitsToBeMasked);
}
