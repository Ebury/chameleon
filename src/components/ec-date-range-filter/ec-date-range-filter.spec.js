import { mount } from '@vue/test-utils';
import { defineComponent } from 'vue';

import { withMockedConsole } from '../../../tests/utils/console';
import EcDateRangeFilter from './ec-date-range-filter.vue';

const label = 'Due date';
const modelValue = { from: new Date(2020, 3, 14), to: new Date(2020, 4, 10) };
const modelValueEmpty = { from: null, to: null };

describe('EcDateRangeFilter', () => {
  function mountEcDateRangeFilter(props, mountOpts) {
    return mount(EcDateRangeFilter, {
      props: {
        fromDatepickerOptions: {
          label: 'From',
        },
        toDatepickerOptions: {
          label: 'To',
        },
        clearText: 'Clear dates',
        modelValue,
        ...props,
      },
      ...mountOpts,
    });
  }

  function mountEcDateRangeFilterAsTemplate(template, props, wrapperComponentOpts, mountOpts) {
    const Component = defineComponent({
      components: { EcDateRangeFilter },
      template,
      ...wrapperComponentOpts,
    });

    return mount(Component, {
      props,
      ...mountOpts,
    });
  }

  it('should render properly when all the props are given', () => {
    const wrapper = mountEcDateRangeFilter({ label });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render with custom attributes', () => {
    const wrapper = mountEcDateRangeFilter({ label }, {
      attrs: {
        'data-test': 'my-data-test',
        class: 'my-class',
        id: 'test-id',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render properly with an error message when the dateRangeErrorMessage is given', () => {
    const wrapper = mountEcDateRangeFilter({ label, dateRangeErrorMessage: 'This is the date range error message' });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render properly with an error message when the fromErrorMessage is given', () => {
    const wrapper = mountEcDateRangeFilter({
      label,
      fromDatepickerOptions: {
        errorMessage: 'This is the from field error message',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render properly with an error message when the toErrorMessage is given', () => {
    const wrapper = mountEcDateRangeFilter({
      label,
      toDatepickerOptions: {
        errorMessage: 'This is the to field error message',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should throw an error if no label prop was given', () => {
    withMockedConsole((errorSpy, warnSpy) => {
      mountEcDateRangeFilter();
      expect(warnSpy).toHaveBeenCalledTimes(1);
      expect(warnSpy.mock.calls[0][0]).toContain('Missing required prop: "label"');
    });
  });

  it('should emit a change event when correct dates are given', async () => {
    const wrapper = mountEcDateRangeFilter({ label });
    await wrapper.findByDataTest('ec-date-range-filter__trigger').trigger('click');
    await wrapper.findByDataTest('ec-date-range-filter__from-input').setValue('2020-11-06');
    await wrapper.findByDataTest('ec-date-range-filter__from-input').trigger('blur');
    expect(wrapper.emitted('change').length).toBe(1);
  });

  it('should emit a change event when the clear dates button is clicked', async () => {
    const wrapper = mountEcDateRangeFilter({ label });
    await wrapper.findByDataTest('ec-date-range-filter__trigger').trigger('click');
    wrapper.findByDataTest('ec-date-range-filter__clear-button').trigger('click');
    expect(wrapper.emitted('change')).toEqual([[{
      from: null,
      to: null,
    }]]);
  });

  it('should not emit a change event when there are no value for dates', async () => {
    const wrapper = mountEcDateRangeFilter({ label, modelValue: modelValueEmpty });
    await wrapper.findByDataTest('ec-date-range-filter__trigger').trigger('click');
    wrapper.findByDataTest('ec-date-range-filter__clear-button').trigger('click');
    expect(wrapper.findByDataTest('ec-date-range-filter__clear-button').attributes('disabled')).toBe('');
    expect(wrapper.emitted('change')).toBeUndefined();
  });

  it('should emit a blur event when the focus is moved away from the date input field', async () => {
    const wrapper = mountEcDateRangeFilter({ label });
    await wrapper.findByDataTest('ec-filter-popover__label').trigger('click');
    await wrapper.findByDataTest('ec-date-range-filter__from-input').trigger('blur');

    expect(wrapper.emitted('blur').length).toBe(1);
  });

  it('should update numberOfSelectedFilters when the dates are passed', async () => {
    const wrapper = mountEcDateRangeFilterAsTemplate(
      '<ec-date-range-filter :label="label" v-model="value"/>',
      {},
      {
        data() {
          return { value: modelValueEmpty, label };
        },
      },
    );

    await wrapper.findByDataTest('ec-date-range-filter__trigger').trigger('click');
    expect(wrapper.findByDataTest('ec-badge').exists()).toBe(false);

    await wrapper.findByDataTest('ec-date-range-filter__from-input').setValue('2020-11-06');
    await wrapper.findByDataTest('ec-date-range-filter__from-input').trigger('blur');
    expect(wrapper.findByDataTest('ec-date-range-filter__from-input').element.value).toBe('2020-11-06');
    expect(wrapper.findByDataTest('ec-badge').text()).toBe('1');

    await wrapper.findByDataTest('ec-date-range-filter__to-input').setValue('2020-12-06');
    await wrapper.findByDataTest('ec-date-range-filter__to-input').trigger('blur');
    expect(wrapper.findByDataTest('ec-date-range-filter__to-input').element.value).toBe('2020-12-06');
    expect(wrapper.findByDataTest('ec-badge').text()).toBe('2');
  });

  it('should render if no dates are passed in value prop', () => {
    const wrapper = mountEcDateRangeFilter({ label, modelValue: null });
    expect(wrapper.element).toMatchSnapshot();
  });
});
