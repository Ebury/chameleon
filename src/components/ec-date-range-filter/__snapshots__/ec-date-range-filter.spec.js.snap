// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EcDateRangeFilter should render if no dates are passed in value prop 1`] = `
<div
  class="ec-filter-popover"
  data-test="ec-date-range-filter__trigger"
>
  <ec-stub
    data-test="ec-stub ec-popover-stub"
    distance="8"
    placement="bottom-start"
    prevent-overflow="false"
  >
    
    <div
      class="ec-filter-popover__trigger"
      data-test="ec-filter-popover__trigger"
    >
      <button
        class="ec-filter-popover__label"
        data-test="ec-filter-popover__label"
        title="Due date"
        type="button"
      >
        Due date
      </button>
      <!--v-if-->
      <svg
        class="ec-icon ec-icon--interactive ec-filter-popover__icon"
        height="16"
        width="16"
      >
        <use
          xlink:href="#ec-simple-chevron-down"
        />
      </svg>
    </div>
    
    
    <div
      class="ec-filter-popover__filter-content"
    >
      
      <div
        class="ec-date-range-filter"
        data-test="ec-date-range-filter"
      >
        <div
          class="ec-date-range-filter__inputs-wrapper"
        >
          <div
            class="ec-input-field ec-date-range-filter__from-input"
            data-test="ec-input-field"
          >
            <label
              class="ec-input-field__label"
              data-test="ec-input-field__label"
              for="ec-input-field-1"
            >
              <span
                class="ec-input-field__label-text"
                data-test="ec-input-field__label-text"
              >
                From
                <!--v-if-->
              </span>
              <!--v-if-->
            </label>
            <input
              class="ec-input-field__input"
              data-test="ec-date-range-filter__from-input ec-input-field__input"
              id="ec-input-field-1"
              placeholder="dd/mm/yyyy"
              type="date"
            />
            <!--v-if-->
            <!--v-if-->
            <!--v-if-->
          </div>
          <div
            class="ec-input-field ec-date-range-filter__to-input"
            data-test="ec-input-field"
          >
            <label
              class="ec-input-field__label"
              data-test="ec-input-field__label"
              for="ec-input-field-2"
            >
              <span
                class="ec-input-field__label-text"
                data-test="ec-input-field__label-text"
              >
                To
                <!--v-if-->
              </span>
              <!--v-if-->
            </label>
            <input
              class="ec-input-field__input"
              data-test="ec-date-range-filter__to-input ec-input-field__input"
              id="ec-input-field-2"
              placeholder="dd/mm/yyyy"
              type="date"
            />
            <!--v-if-->
            <!--v-if-->
            <!--v-if-->
          </div>
        </div>
        <!--v-if-->
        <button
          class="ec-date-range-filter__clear-button"
          data-test="ec-date-range-filter__clear-button"
          disabled=""
          type="button"
        >
          Clear dates
        </button>
      </div>
      
    </div>
    
  </ec-stub>
</div>
`;

exports[`EcDateRangeFilter should render properly when all the props are given 1`] = `
<div
  class="ec-filter-popover"
  data-test="ec-date-range-filter__trigger"
>
  <ec-stub
    data-test="ec-stub ec-popover-stub"
    distance="8"
    placement="bottom-start"
    prevent-overflow="false"
  >
    
    <div
      class="ec-filter-popover__trigger"
      data-test="ec-filter-popover__trigger"
    >
      <button
        class="ec-filter-popover__label"
        data-test="ec-filter-popover__label"
        title="Due date"
        type="button"
      >
        Due date
      </button>
      <span
        class="ec-badge ec-badge--info tw-ml-4"
        data-test="ec-badge"
      >
        
        2
        
      </span>
      <svg
        class="ec-icon ec-icon--interactive ec-filter-popover__icon"
        height="16"
        width="16"
      >
        <use
          xlink:href="#ec-simple-chevron-down"
        />
      </svg>
    </div>
    
    
    <div
      class="ec-filter-popover__filter-content"
    >
      
      <div
        class="ec-date-range-filter"
        data-test="ec-date-range-filter"
      >
        <div
          class="ec-date-range-filter__inputs-wrapper"
        >
          <div
            class="ec-input-field ec-date-range-filter__from-input"
            data-test="ec-input-field"
          >
            <label
              class="ec-input-field__label"
              data-test="ec-input-field__label"
              for="ec-input-field-1"
            >
              <span
                class="ec-input-field__label-text"
                data-test="ec-input-field__label-text"
              >
                From
                <!--v-if-->
              </span>
              <!--v-if-->
            </label>
            <input
              class="ec-input-field__input"
              data-test="ec-date-range-filter__from-input ec-input-field__input"
              id="ec-input-field-1"
              max="2020-04-10"
              placeholder="dd/mm/yyyy"
              type="date"
            />
            <!--v-if-->
            <!--v-if-->
            <!--v-if-->
          </div>
          <div
            class="ec-input-field ec-date-range-filter__to-input"
            data-test="ec-input-field"
          >
            <label
              class="ec-input-field__label"
              data-test="ec-input-field__label"
              for="ec-input-field-2"
            >
              <span
                class="ec-input-field__label-text"
                data-test="ec-input-field__label-text"
              >
                To
                <!--v-if-->
              </span>
              <!--v-if-->
            </label>
            <input
              class="ec-input-field__input"
              data-test="ec-date-range-filter__to-input ec-input-field__input"
              id="ec-input-field-2"
              min="2020-03-14"
              placeholder="dd/mm/yyyy"
              type="date"
            />
            <!--v-if-->
            <!--v-if-->
            <!--v-if-->
          </div>
        </div>
        <!--v-if-->
        <button
          class="ec-date-range-filter__clear-button"
          data-test="ec-date-range-filter__clear-button"
          type="button"
        >
          Clear dates
        </button>
      </div>
      
    </div>
    
  </ec-stub>
</div>
`;

exports[`EcDateRangeFilter should render properly with an error message when the dateRangeErrorMessage is given 1`] = `
<div
  class="ec-filter-popover"
  data-test="ec-date-range-filter__trigger"
>
  <ec-stub
    data-test="ec-stub ec-popover-stub"
    distance="8"
    placement="bottom-start"
    prevent-overflow="false"
  >
    
    <div
      class="ec-filter-popover__trigger"
      data-test="ec-filter-popover__trigger"
    >
      <button
        class="ec-filter-popover__label"
        data-test="ec-filter-popover__label"
        title="Due date"
        type="button"
      >
        Due date
      </button>
      <span
        class="ec-badge ec-badge--info tw-ml-4"
        data-test="ec-badge"
      >
        
        2
        
      </span>
      <svg
        class="ec-icon ec-icon--interactive ec-filter-popover__icon"
        height="16"
        width="16"
      >
        <use
          xlink:href="#ec-simple-chevron-down"
        />
      </svg>
    </div>
    
    
    <div
      class="ec-filter-popover__filter-content"
    >
      
      <div
        class="ec-date-range-filter"
        data-test="ec-date-range-filter"
      >
        <div
          class="ec-date-range-filter__inputs-wrapper"
        >
          <div
            class="ec-input-field ec-date-range-filter__from-input"
            data-test="ec-input-field"
          >
            <label
              class="ec-input-field__label"
              data-test="ec-input-field__label"
              for="ec-input-field-1"
            >
              <span
                class="ec-input-field__label-text"
                data-test="ec-input-field__label-text"
              >
                From
                <!--v-if-->
              </span>
              <!--v-if-->
            </label>
            <input
              class="ec-input-field__input"
              data-test="ec-date-range-filter__from-input ec-input-field__input"
              id="ec-input-field-1"
              max="2020-04-10"
              placeholder="dd/mm/yyyy"
              type="date"
            />
            <!--v-if-->
            <!--v-if-->
            <!--v-if-->
          </div>
          <div
            class="ec-input-field ec-date-range-filter__to-input"
            data-test="ec-input-field"
          >
            <label
              class="ec-input-field__label"
              data-test="ec-input-field__label"
              for="ec-input-field-2"
            >
              <span
                class="ec-input-field__label-text"
                data-test="ec-input-field__label-text"
              >
                To
                <!--v-if-->
              </span>
              <!--v-if-->
            </label>
            <input
              class="ec-input-field__input"
              data-test="ec-date-range-filter__to-input ec-input-field__input"
              id="ec-input-field-2"
              min="2020-03-14"
              placeholder="dd/mm/yyyy"
              type="date"
            />
            <!--v-if-->
            <!--v-if-->
            <!--v-if-->
          </div>
        </div>
        <p
          class="ec-date-range-filter__error-text"
          data-test="ec-date-range-filter__error-text"
        >
          This is the date range error message
        </p>
        <button
          class="ec-date-range-filter__clear-button"
          data-test="ec-date-range-filter__clear-button"
          type="button"
        >
          Clear dates
        </button>
      </div>
      
    </div>
    
  </ec-stub>
</div>
`;

exports[`EcDateRangeFilter should render properly with an error message when the fromErrorMessage is given 1`] = `
<div
  class="ec-filter-popover"
  data-test="ec-date-range-filter__trigger"
>
  <ec-stub
    data-test="ec-stub ec-popover-stub"
    distance="8"
    placement="bottom-start"
    prevent-overflow="false"
  >
    
    <div
      class="ec-filter-popover__trigger"
      data-test="ec-filter-popover__trigger"
    >
      <button
        class="ec-filter-popover__label"
        data-test="ec-filter-popover__label"
        title="Due date"
        type="button"
      >
        Due date
      </button>
      <span
        class="ec-badge ec-badge--info tw-ml-4"
        data-test="ec-badge"
      >
        
        2
        
      </span>
      <svg
        class="ec-icon ec-icon--interactive ec-filter-popover__icon"
        height="16"
        width="16"
      >
        <use
          xlink:href="#ec-simple-chevron-down"
        />
      </svg>
    </div>
    
    
    <div
      class="ec-filter-popover__filter-content"
    >
      
      <div
        class="ec-date-range-filter"
        data-test="ec-date-range-filter"
      >
        <div
          class="ec-date-range-filter__inputs-wrapper"
        >
          <div
            class="ec-input-field ec-date-range-filter__from-input"
            data-test="ec-input-field"
          >
            <label
              class="ec-input-field__label"
              data-test="ec-input-field__label"
              for="ec-input-field-1"
            >
              <span
                class="ec-input-field__label-text"
                data-test="ec-input-field__label-text"
              >
                From
                <!--v-if-->
              </span>
              <!--v-if-->
            </label>
            <input
              aria-describedby="ec-input-field-error-1"
              class="ec-input-field__input ec-input-field__input--has-error"
              data-test="ec-date-range-filter__from-input ec-input-field__input"
              id="ec-input-field-1"
              max="2020-04-10"
              placeholder="dd/mm/yyyy"
              type="date"
            />
            <!--v-if-->
            <!--v-if-->
            <div
              class="ec-input-field__error-text"
              data-test="ec-input-field__error-text"
              id="ec-input-field-error-1"
            >
              This is the from field error message
            </div>
          </div>
          <div
            class="ec-input-field ec-date-range-filter__to-input"
            data-test="ec-input-field"
          >
            <label
              class="ec-input-field__label"
              data-test="ec-input-field__label"
              for="ec-input-field-2"
            >
              <span
                class="ec-input-field__label-text"
                data-test="ec-input-field__label-text"
              >
                To
                <!--v-if-->
              </span>
              <!--v-if-->
            </label>
            <input
              class="ec-input-field__input"
              data-test="ec-date-range-filter__to-input ec-input-field__input"
              id="ec-input-field-2"
              min="2020-03-14"
              placeholder="dd/mm/yyyy"
              type="date"
            />
            <!--v-if-->
            <!--v-if-->
            <!--v-if-->
          </div>
        </div>
        <!--v-if-->
        <button
          class="ec-date-range-filter__clear-button"
          data-test="ec-date-range-filter__clear-button"
          type="button"
        >
          Clear dates
        </button>
      </div>
      
    </div>
    
  </ec-stub>
</div>
`;

exports[`EcDateRangeFilter should render properly with an error message when the toErrorMessage is given 1`] = `
<div
  class="ec-filter-popover"
  data-test="ec-date-range-filter__trigger"
>
  <ec-stub
    data-test="ec-stub ec-popover-stub"
    distance="8"
    placement="bottom-start"
    prevent-overflow="false"
  >
    
    <div
      class="ec-filter-popover__trigger"
      data-test="ec-filter-popover__trigger"
    >
      <button
        class="ec-filter-popover__label"
        data-test="ec-filter-popover__label"
        title="Due date"
        type="button"
      >
        Due date
      </button>
      <span
        class="ec-badge ec-badge--info tw-ml-4"
        data-test="ec-badge"
      >
        
        2
        
      </span>
      <svg
        class="ec-icon ec-icon--interactive ec-filter-popover__icon"
        height="16"
        width="16"
      >
        <use
          xlink:href="#ec-simple-chevron-down"
        />
      </svg>
    </div>
    
    
    <div
      class="ec-filter-popover__filter-content"
    >
      
      <div
        class="ec-date-range-filter"
        data-test="ec-date-range-filter"
      >
        <div
          class="ec-date-range-filter__inputs-wrapper"
        >
          <div
            class="ec-input-field ec-date-range-filter__from-input"
            data-test="ec-input-field"
          >
            <label
              class="ec-input-field__label"
              data-test="ec-input-field__label"
              for="ec-input-field-1"
            >
              <span
                class="ec-input-field__label-text"
                data-test="ec-input-field__label-text"
              >
                From
                <!--v-if-->
              </span>
              <!--v-if-->
            </label>
            <input
              class="ec-input-field__input"
              data-test="ec-date-range-filter__from-input ec-input-field__input"
              id="ec-input-field-1"
              max="2020-04-10"
              placeholder="dd/mm/yyyy"
              type="date"
            />
            <!--v-if-->
            <!--v-if-->
            <!--v-if-->
          </div>
          <div
            class="ec-input-field ec-date-range-filter__to-input"
            data-test="ec-input-field"
          >
            <label
              class="ec-input-field__label"
              data-test="ec-input-field__label"
              for="ec-input-field-2"
            >
              <span
                class="ec-input-field__label-text"
                data-test="ec-input-field__label-text"
              >
                To
                <!--v-if-->
              </span>
              <!--v-if-->
            </label>
            <input
              aria-describedby="ec-input-field-error-2"
              class="ec-input-field__input ec-input-field__input--has-error"
              data-test="ec-date-range-filter__to-input ec-input-field__input"
              id="ec-input-field-2"
              min="2020-03-14"
              placeholder="dd/mm/yyyy"
              type="date"
            />
            <!--v-if-->
            <!--v-if-->
            <div
              class="ec-input-field__error-text"
              data-test="ec-input-field__error-text"
              id="ec-input-field-error-2"
            >
              This is the to field error message
            </div>
          </div>
        </div>
        <!--v-if-->
        <button
          class="ec-date-range-filter__clear-button"
          data-test="ec-date-range-filter__clear-button"
          type="button"
        >
          Clear dates
        </button>
      </div>
      
    </div>
    
  </ec-stub>
</div>
`;
